### DISCUSSION TITLE
Event after sorting

### DISCUSSION THREAD

**Discussion by WebVPF** [wintercms/winter#691] [Q&A] [answered]
The model has trait Sortable `use \Winter\Storm\Database\Traits\Sortable;`

After sorting I need to delete the cache

```php
\Cache::forget('menu_items');
```

How can I do it?

**Comment by RomainMazB** [comment]
What about using the model events to look if the `$model->getSortOrderColumn()` changed?
Something like
```php
class MyModel extends Model {
  public $implements = ['Winter.Storm.Database.Traits.Sortable'];

  public function afterSave() {
    if ($this->wasChanged($this->getSortOrderColumn())
    {
      \Cache::forget('menu_items');
    }
  }
}
```

Not a perfect solution since it will be fired many times in a single request though (because many model are saved when reordering)

**Comment by WebVPF** [comment, marked as answer]
https://github.com/wintercms/winter/issues/952#issuecomment-1651760550
